pkg/cmd/backuprestore/crioclient.go

```diff
@@ -0,0 +1,117 @@
+package backuprestore
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"k8s.io/klog"
+	"net"
+	"net/url"
+	"time"
+
+	pb "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+const (
+	defaultTimeout = 2 * time.Second
+	unixProtocol   = "unix"
+)
+
+// Code taken from github.com/kubernetes-sigs/cri-tools/cmd/crictl
+
+var defaultRuntimeEndpoint string = "unix:///run/crio/crio.sock"
+
+func getRuntimeCrioClient() (pb.RuntimeServiceClient, *grpc.ClientConn, error) {
+	// Set up a connection to the server.
+	conn, err := getRuntimeClientConnection()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "connect")
+	}
+	runtimeClient := pb.NewRuntimeServiceClient(conn)
+	return runtimeClient, conn, nil
+}
+
+func closeConnection(conn *grpc.ClientConn) error {
+	if conn == nil {
+		return nil
+	}
+	return conn.Close()
+}
+
+func getRuntimeClientConnection() (*grpc.ClientConn, error) {
+	return getConnection(defaultRuntimeEndpoint)
+}
+
+func getConnection(endPoint string) (*grpc.ClientConn, error) {
+	var conn *grpc.ClientConn
+	klog.Infof("connect using endpoint '%s' with '%s' timeout", endPoint, defaultTimeout)
+	addr, dialer, err := getAddressAndDialer(endPoint)
+	if err != nil {
+		return nil, err
+	}
+	conn, err = grpc.Dial(addr, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithTimeout(defaultTimeout), grpc.WithContextDialer(dialer))
+	if err != nil {
+		errMsg := errors.Wrapf(err, "connect endpoint '%s', make sure you are running as root and the endpoint has been started", endPoint)
+		return nil, errMsg
+	} else {
+		klog.Infof("connected successfully using endpoint: %s", endPoint)
```

Comment:
No need to log success, the endpoint attempt details are already logged above

Reply:
Done.